webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +87,1137 @@ public WebSiteSessionController createComponentSessionController(
+        request.setAttribute("AllIcons", scc.getAllIcons());
+        request.setAttribute("Action", "suggest");
+        destination = "/webSites/jsp/descBookmark.jsp";
+      } else if (function.startsWith("descBookmark.jsp")) {
+        request.setAttribute("AllIcons", scc.getAllIcons());
+        destination = "/webSites/jsp/descBookmark.jsp";
+      } else if (function.startsWith("descDesign.jsp")) {
+        request.setAttribute("AllIcons", scc.getAllIcons());
+        destination = "/webSites/jsp/descDesign.jsp";
+      } else if (function.startsWith("organize.jsp")) {
+        destination = organizeSites(scc, request, flag);
+      } else if ("AddTopic".equals(function)) {
+        destination = addTopic(scc, request, flag);
+      } else if (function.startsWith("updateTopic")) {
+        destination = updateTopic(scc, request, flag);
+      } else if (function.startsWith("classifyDeclassify.jsp")) {
+        destination = classification(scc, request, flag);
+      } else if (function.startsWith("manage.jsp")) {
+        destination = manageSite(scc, request, flag);
+      } else if (function.startsWith("design.jsp")) {
+        destination = designSite(scc, request, flag);
+      } else if (function.equals("EffectiveUploadFile")) {
+        destination = uploadFile(scc, request, flag);
+      } else if (function.startsWith("descUpload.jsp")) {
+        request.setAttribute("AllIcons", scc.getAllIcons());
+        destination = "/webSites/jsp/descUpload.jsp";
+      } else if (function.equals("EffectiveUploadSiteZip")) {
+        destination = uploadSite(scc, request, flag);
+      } else {
+        destination = "/webSites/jsp/" + function;
+      }
+    } catch (Exception e) {
+      request.setAttribute("javax.servlet.jsp.jspException", e);
+      destination = "/admin/jsp/errorpageMain.jsp";
+    }
 
-        if (action == null) {
-          id = "0";
-          action = "Search";
-        }
+    return destination;
+  }
 
-        FolderDetail webSitesCurrentFolder = scc.getFolder(id);
-        scc.setSessionTopic(webSitesCurrentFolder);
-        request.setAttribute("CurrentFolder", webSitesCurrentFolder);
+  private String uploadSite(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws Exception {
+    List<FileItem> items = request.getFileItems();
 
-        destination = "/webSites/jsp/listSite.jsp?Action=" + action + "&Id=" + id;
-      } else if (function.startsWith("listSite_reader.jsp")) {
-        String action = request.getParameter("Action");
-        String id = request.getParameter("Id");
+    String nomSite = FileUploadUtil.getParameter(items, "nomSite");
+    String description = FileUploadUtil.getParameter(items, "description");
+    String popupString = FileUploadUtil.getParameter(items, "popup");
+    int popup = 0;
+    if ("on".equals(popupString)) {
+      popup = 1;
+    }
+    String nomPage = FileUploadUtil.getParameter(items, "nomPage");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "nomPage" 7 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)